### ISSUE TITLE
Not working correctly with umbrella projects

### ISSUE THREAD

**Issue by WindDrifter** [hauleth/mix_unused#28]
I tried using this with an umbrella project there are issues that I found. Note that I ran `mix compile` at root of project

Issue one:
put `compilers: [:unused] ++ Mix.compilers()` in the root mix.exs file will generate 0 result (I purposely put a unused function in one of the sub project that is inside the umbrella)

Issue two
put `compilers: [:unused] ++ Mix.compilers()` in all sub projects' `mix.exs` file will generate false postivies (ie subproject one function will be deem unused even though it is used by subproject two)

**Comment by denispeplin** [comment]
Same for me. I tried it both ways, one way it doesn't show anything, another way it shows false positives.

**Comment by hauleth** [comment]
Yes, this is known limitation of the project at the moment. Reason for that is simple:

> In umbrella there is no distinction between other project in the umbrella and regular dependency.

You need to remember that at it's core umbrella projects are no different from other projects that use `{:my_dep, path: "../my_dep"}`. So while it is welcome improvement, I do not have interest in working on it personally, as in my code book each application in umbrella should be considered as a standalone project, that by the convenience is stored in one repository.

**Comment by denispeplin** [comment]
> in my code book each application in umbrella should be considered as a standalone project, that by the convenience is stored in one repository

So maybe it's for some distant future. I can tell it from my experience: for an umbrella project that several years old and being built by dozens of people, having unused function detection would be quite valuable.

**Comment by hauleth** [comment]
Well, if someone is willing to spend time on such feature then I am willing to review and help with implementation, but even if I would have use for such feature, currently I do not have time to work on it. Day to day job and kid is too absorbing to have even moment to spare.

I can guide potential implementor though.
